roles/pulp_database/tasks/install_postgres.yml

```diff
@@ -1,29 +1,111 @@
 ---
 - block:
+    - name: Check if PostgreSQL < 10 is initialized.
+      stat:
+        path: "{{ postgresql_previous_data_dir }}/PG_VERSION"
+      register: previous_pgdata_dir_version
+
+    - name: Get PG_VERSION
+      command: "cat {{ postgresql_previous_data_dir }}/PG_VERSION"
+      register: pg_version
+      when: previous_pgdata_dir_version.stat.exists
+      tags:
+        - molecule-idempotence-notest
+
+    - name: Stop PostgreSQL < 10 service
+      systemd:
+        name: "{{ postgresql_previous_daemon }}"
+        state: stopped
+        enabled: no
+      register: pgservice_result
+      failed_when: "pgservice_result is failed and 'Service is in unknown state' not in pgservice_result.msg"
+      when:
+        - previous_pgdata_dir_version.stat.exists
+        - pg_version is defined
+        - pg_version.changed
+        - pg_version.stdout is version('10', '<')
+
     - name: Install SCL package
       package:
-        name: 'centos-release-scl'
+        name:
+          - 'scl-utils'
+          - 'centos-release-scl-rh'
+          - 'centos-release-scl'
         state: present
-      when:
-        - ansible_facts.distribution == 'CentOS'
-        - ansible_facts.distribution_major_version|int == 7
 
     - name: Install PostgreSQL SCL
-      package:
-        name: 'rh-postgresql96'
+      yum:
+        name: "{{ postgresql_packages }}"
         state: present
-      when:
-        - ansible_facts.distribution == 'CentOS'
-        - ansible_facts.distribution_major_version|int == 7
+        exclude: python-unversioned-command
 
     - name: Enable PostgreSQL SCL
       template:
         src: templates/postgresql_scl_profile.j2
         dest: "{{ postgresql_profile_path }}"
+
+    - name: Create SCL symlinks
+      file:
+        src: "{{ item }}"
+        dest: "{{ item.replace('/opt/rh/rh-postgresql10/root', '') }}"
+        state: link
+      loop:
+        - "/opt/rh/rh-postgresql10/root/usr/lib64/libpq.so.rh-postgresql10-5"
+        - "/opt/rh/rh-postgresql10/root/usr/bin/pg_config"
+      register: symlink_result
+      failed_when: "symlink_result is failed and 'refusing to convert from file to symlink' not in symlink_result.msg"
+      check_mode: false
+
+    - name: Backup previous PostgreSQL
```

Comment:
Nice. I assume you are going off of the instructions in the RH SCL docs?

It would be helpful to list the URL as a comment.